### PR TITLE
Devops tests user

### PR DESCRIPTION
**Pull request by diogo-botelho**
Added tests for user model with 100% coverage.
* Finished tests for user model: 100% coverage
* Updated schema to cascade delete user and busker
* Created jest coverage report.

### CODE REVIEW

File: backend/src/models/user.ts [hunk 1]
```diff
@@ -16,6 +16,16 @@ import {
 import { BCRYPT_WORK_FACTOR } from "../config";
 import { UserData } from "../interfaces/UserData";
 
+interface updateUserData {
```
Comment (samau3) [inline backend/src/models/user.ts]: Capitalize "u" so it's `UpdateUserData`

File: backend/src/models/user.ts [hunk 2]
```diff
@@ -165,7 +182,12 @@ export class User {
    * or a serious security risks are opened.
    * */
 
-  static async update(username: string, data: UserData): Promise<UserData> {
+  static async update(
+    username: string,
+    data: updateUserData
```
Comment (samau3) [inline backend/src/models/user.ts]: Capitalize "u" so it's `UpdateUserData`

File: backend/src/models/user.ts [hunk 3]
```diff
@@ -175,11 +197,12 @@ export class User {
...
       isAdmin: "is_admin",
     });
-    const idVarIdx = "$" + (values.length + 1);
+
+    const usernameVarIdx = "$" + (values.length + 1);
 
     const querySql = `UPDATE users
             SET ${setCols} 
-            WHERE id = ${idVarIdx} 
+            WHERE username = ${usernameVarIdx}
```
Comment (samau3) [inline backend/src/models/user.ts:185]: If a user is updating their username in the same request, wouldn't this cause an error as the `WHERE` condition won't be able to find an existing user with the updated username?

**Review by samau3 — COMMENTED** [review]
Some minor questions and a question!

File: backend/src/models/user.ts (hunk 3 above)
Comment (diogo-botelho) [reply backend/src/models/user.ts:185]: Resolved - usernameVarIdx is going to be used to match `WHERE username = ${usernameVarIdx}` on line 182 to the username variable on line 190.

File: backend/src/models/user.ts (hunk 2 above)
Comment (diogo-botelho) [reply backend/src/models/user.ts]: Resolved - deleted  UpdateUserData interface, using UserData interface across the user model instead.

File: backend/src/models/user.ts (hunk 1 above)
Comment (diogo-botelho) [reply backend/src/models/user.ts]: Same answer as: https://github.com/diogo-botelho/busk-app/pull/27#discussion_r1148089076

**Pull request merged**